field/FieldPointerUtilities.F90

```diff
@@ -877,8 +880,34 @@ subroutine MAPL_FieldGetLocalElementCount(field,local_count,rc)
          else if (rank ==4) then
             call ESMF_FieldGet(field,0,farrayptr=r8_4d,_RC)
             local_count = shape(r8_4d)
-         else
-            _FAIL("Unsupported rank")
+         end if
+      else if (tk == ESMF_TypeKind_I4) then
+         if (rank==1) then
+            call ESMF_FieldGet(field,0,farrayptr=i4_1d,_RC)
+            local_count = shape(i4_1d)
+         else if (rank ==2) then
+            call ESMF_FieldGet(field,0,farrayptr=i4_2d,_RC)
+            local_count = shape(i4_2d)
+         else if (rank ==3) then
+            call ESMF_FieldGet(field,0,farrayptr=i4_3d,_RC)
+            local_count = shape(i4_3d)
+         else if (rank ==4) then
+            call ESMF_FieldGet(field,0,farrayptr=i4_4d,_RC)
+            local_count = shape(i4_4d)
+         end if
```

Comment:
Please replace with SELECT CASE construct.   If nothing else this will make it more obvious what happens if rank is 0 or larger than 4 because it will have a DEFAULT block.

Same for block of IFs below.